### ISSUE TITLE
iOS 14 compatibility

### ISSUE THREAD

**Issue by defagos** [SRGSSR/playsrg-apple#157]
We must ensure Play runs correctly on iOS 14. Issues to be fixed and behavior to verify:

- [x] Broken timeline layout each time it is opened (the bug happens in Play only, not in Letterbox demo).
- [x] No permission issues are encountered (or if there are any, they are properly dealt with).
- [x] Google Cast works fine.
- [x] Player rotation works well with PiP and does not yield usually unsupported layouts.
- [x] The song panel is correctly displayed and behaves properly (not seen on my iPhone 7 device yet, but seen in the simulator).
- [x] Background video playbacks is restored or removed.
- [x] Check that remote configuration works correctly (Firebase requires an update for Protobuf compatibility with Google Cast SDK).

**Comment by defagos** [comment]
With the latest beta 3, we have a player leak when dismissing the player on iOS with our interactive gesture, probably related to PiP somehow.

The same issue does not seem to affect Letterbox demo, but I am not sure the issue is entirely on our side (it worked fine on iPad where PiP was available since a long time).

I could not see any other major issues. I am also in the process of updating to SPM, and our library UTs seemed to work fine with beta 2.

**Comment by defagos** [comment]
I also got the leak with `AVPlayerViewController`, so I assume this is an iOS 14 beta bug. The new beta 3 also broke media playback again in the simulator ☹️

**Comment by defagos** [comment]
I saw a small issue (probably an iOS bug): With the player playing some content, lock the device and unlock it. The progress bar is then tinted in blue on the left side of its knob.

**Comment by defagos** [comment]
The Google Cast SDK [must be updated to version 4.5.0](https://developers.google.com/cast/docs/ios_sender/ios_permissions_changes) once available.

**Comment by defagos** [comment]
Apple took everybody by surprise by announcing the iOS 14 release. The PiP leak is fixed when building and running with the GM and updates for libraries we use are apparently available (including Google Cast SDK).

I'll start preparting an update so that we can test it internally. We probably still have to plan a delivery sooner than expected, as existing apps built with the iOS 13 SDK suffer from the PiP leak, forcing users to quit the app.

**Comment by pyby** [comment]
Broken timeline layout each time it is opened, fixed with Letterbox 5.0.2 release (see [commit](https://github.com/SRGSSR/srgletterbox-apple/commit/bc6f6be51acf2305f29a76953af547cd9ce31ffd)).

**Comment by defagos** [comment]
The Google Cast SDK issue has been addressed. When compiled with the iOS 14 SDK, the app applies new local network permissions. Version 4.5.0 of the Google Cast SDK implements the needed support, which still [requires a few code changes](https://developers.google.com/cast/docs/ios_sender/ios_permissions_changes).

In particular, the receiver identifier must be declared under the `NSBonjourServices` key of the Info.plist file. It seems discovery works if only `_googlecast._tcp` is declared, but I wouldn't be this works in all possible discovery scenarios. For this reason, the receiver identifier cannot be remotely changed anymore (we could maybe declare more known identifiers in the Info.plist, but again the documentation does not mention what happens in this case). Since this is rather the exception than the rule, I removed the corresponding remote configuration parameter and hardcoded it in the project settings.

Moreover I updated the Google Cast view appearance (fonts and colors). Fonts won't adjust once the app has been launched (the Google Cast views are cached and reused), still we can start with the current system font size, which should cover most needs appropriately.

Finally, if the user refuses local network access the first time a request onto the local network is made, there is no way to trigger the corresponding alert again. For this reason, the system settings button has been moved to a dedicated _Manage permissions_ section. This will be helpful if users contact support because they refused permissions but don't know why they cannot see their Google Cast receiver anymore.
